doc/cascadia/JSONFragments.md

```diff
@@ -0,0 +1,123 @@
+
+Of course, if the developer only wishes to add/modify profiles without adding color schemes (and vice-versa), only the relevant list needs to be present and the other list can be omitted.
+
+## How to determine the GUID of an existing profile 
+
+The only profiles that can be modified through fragments are the default profiles, Command Prompt and PowerShell, as well as [dynamic profiles](https://docs.microsoft.com/windows/terminal/dynamic-profiles). To determine the GUID of the profile to be updated, use a Version 5 UUID generator with the following namespace GUID and name:
+
+- The namespace GUID: { 2BDE4A90 - D05F - 401C - 9492 - E40884EAD1D8 } 
+- The name of the profile to be updated 
+
+As a sanity check, a profile called 'Ubuntu' will get the generated GUID: { 2C4DE342 - 38B7 - 51CF - B940 - 2309A097F518 } 
+
+## Minimum requirements for settings added with fragments 
+
+There are some minimal restrictions on what can be added to user settings using JSON fragments:
+
+- For new profiles added via fragments, the new profile must, at a minimum, define a name for itself.
+- For new color schemes added via fragments, the new color scheme must define a name for itself, as well as define every color in the color table (i.e. the colors "black" through "brightYellow" in the example image above).
+
+## Where to place the JSON fragment files
+
+The location to place the JSON fragment files varies depending on the installation method of the application that wishes to place them.  
+
+### Microsoft Store applications 
+
+For applications installed through the Microsoft Store (or similar), the application must declare itself to be an app extension. More details on app extensions can be found in the [Microsoft Docs](https://docs.microsoft.com/windows/uwp/launch-resume/how-to-create-an-extension) and the necessary section is replicated here. The appxmanifest file of the package must include: 
+
+```xml
+<Package
+  ...
+  xmlns:uap3="http://schemas.microsoft.com/appx/manifest/uap/windows10/3"
+  IgnorableNamespaces="uap uap3 mp">
+  ...
+    <Applications>
+      <Application Id="App" ... >
+        ...
+        <Extensions>
+          ...
+          <uap3:Extension Category="windows.appExtension">
+            <uap3:AppExtension Name="com.microsoft.windows.terminal.settings"
+                               Id="<id>"
+                               PublicFolder="Public">
+            </uap3:AppExtension>
+          </uap3:Extension>
+        </Extensions>
+      </Application>
+    </Applications>
+    ...
+</Package>
+```
+
+Key things to note: 
+
+- The `"Name"` field must be `com.microsoft.windows.terminal.settings` for Windows Terminal to be able to detect the extension.
+- The `"Id"` field can be filled out as the developer wishes.
```

Comment:
I forgot if we implemented this so @PankajBhojwani back me up:

should we mention that this Id is something that end-users can use to disable these profiles with `disabledProfileSources`?

Reply:
Ah, the `source` field that we assign to these type of apps is actually `Id().FamilyName()` which is actually some weird thing with a bunch of numbers in it that doesn't seem to have anything to do with just the `Id`